src/convertToObject.js

```diff
@@ -3,16 +3,26 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const STYLES_OBJECT = sourceString
+    .replace(/\s+/g, ' ')
+    .replace(/\s:/g, ':')
+    .split(';')
+    .map((prop) => prop.trim())
+    .filter((prop) => prop)
+    .map(prop => prop.split(': '));
```

Comment:
I like the solution with regex, but 3 loops one after another is too much, remember about efficiency of code. Try combining it into one loop (Hint: forEach or reduce)

Reply:
thank you! I tried my best). Fixed according to your hints ;)